providers/com/philica/normalizer.py

```diff
@@ -0,0 +1,95 @@
+import arrow
+
+import dateparser
+
+from share.normalize import Parser, Delegate, RunPython, ParseName, Normalizer, Concat, Map, ctx, Try
+
+
+class Link(Parser):
+    url = ctx
+    type = RunPython('get_link_type', ctx)
+
+    def get_link_type(self, link):
+        if 'dx.doi.org' in link:
+            return 'doi'
+        if 'philica.com' in link:
+            return 'provider'
+        return 'misc'
+
+
+class ThroughLinks(Parser):
+    link = Delegate(Link, ctx)
+
+
+class Publisher(Parser):
+    name = ctx
+
+
+class Tag(Parser):
+    name = ctx
+
+
+class ThroughTags(Parser):
+    tag = Delegate(Tag, ctx)
+
+
+class Association(Parser):
+    entity = Delegate(Publisher)
+
+
+class Person(Parser):
+    given_name = ParseName(ctx).first
+    family_name = ParseName(ctx).last
+
+
+class Contributor(Parser):
+    person = Delegate(Person, ctx)
+    cited_name = ctx
+    order_cited = ctx('index')
+
+
+class Preprint(Parser):
+    title = Try(ctx['DC.title'])
+    description = Try(ctx['DC.description'])
+    contributors = Map(Delegate(Contributor), ctx['DC.contributor'])
+    links = Map(Delegate(ThroughLinks),
+            ctx['href'],
+            ctx['DC.source']
+    )
+    publishers = Map(
+        Delegate(Association),
```

Comment:
This can be updated to `Delegate(Association.using(entity=Delegate(Publisher))),` and then the `Association` class can be used for multiple types of associations.